crates/stackable-telemetry/src/tracing/settings/mod.rs

```diff
@@ -0,0 +1,166 @@
+//! Subscriber settings.
+
+use tracing::level_filters::LevelFilter;
+
+pub mod console_log;
+pub use console_log::*;
+
+pub mod otlp_log;
+pub use otlp_log::*;
+
+pub mod otlp_trace;
+pub use otlp_trace::*;
+
+/// Simplifies access common settings from subscriber specific settings.
+pub trait CommonSettings {
+    /// Access to the [`Settings::environment_variable`] field.
+    fn environment_variable(&self) -> &'static str;
+
+    /// Access to the [`Settings::default_level`] field.
+    fn default_level(&self) -> LevelFilter;
+
+    /// Access to the [`Settings::enabled`] field.
+    fn enabled(&self) -> bool;
```

Comment:
Would a disabled subscriber even exist beyond the builder stage?

Reply:
No, but we still need access to the `enabled` field, which is part of the common `Settings` struct.

Reply:
I'm not going to cause a fuss over it right now, but it would be nice to see if we can kill it off eventually.

Reply:
I'm unsure what you mean, can you describe a little more concretely?

The actual Tracing Subscriber/Layer is never created/registered if `enabled` is false.

Reply:
I haven't tested it or thought it through completely, but in short, you could use `bool::then` and let the `with_foo` take an `Option<FooSettings>`. Something like:

```rust
.with_otlp_log_exporter(
  otlp_log_flag.then(|| Settings::builder()
    .blah()
    .build()
  )
)
```

Reply:
I have added an item to the parent issue: https://github.com/stackabletech/issues/issues/639